tests/components/nmbs/test_config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Test the NMBS config flow."""
+
+from copy import deepcopy
+from typing import Any
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.nmbs.const import (
+    CONF_STATION_FROM,
+    CONF_STATION_LIVE,
+    CONF_STATION_TO,
+    DOMAIN,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import mock_api_unavailable, mock_station_response
+
+DUMMY_DATA: dict[str, Any] = {
+    "STAT_BRUSSELS_NORTH": "Brussel-Noord/Bruxelles-Nord",
+    "STAT_BRUSSELS_CENTRAL": "Brussel-Centraal/Bruxelles-Central",
+    "STAT_BRUSSELS_SOUTH": "Brussel-Zuid/Bruxelles-Midi",
+}
+
+
+async def test_step_connection_no_data(hass: HomeAssistant) -> None:
+    """Test starting a flow by user which chooses for connection."""
+    with patch(
+        "pyrail.irail.iRail.get_stations",
```

Comment:
Let's patch where we use this, so
```suggestion
        "homeassistant.components.nmbs.iRail.get_stations",
```